### DISCUSSION TITLE
Partial workflow reruns

### DISCUSSION THREAD

**Discussion by svarshavchik** [community/community#15239] [Actions]
I found the following when I searched for a way to rerun failed jobs in a workflow: https://github.blog/2022-03-16-save-time-partial-re-runs-github-actions/

Unfortunately this does not help when the reason for wanting to rerun a failed job is that I am working on building jobs for new workflow. Rerunning the failed job, as is, will make no difference. I want to push out a fixed job, and rerun the fixed job, have it access to the already-uploaded artifacts, and basically pick up where I failed.

Right now, the initial jobs in my workflow take about twenty minutes to get their work done before getting to the leaf jobs I'm working on. So, every time it fails, I dig it up, figure out why, and either fix it, or hope that I fix it, I have to wait more minutes to see if the fix worked.

So, to summarize:

1) The initial jobs in my workflow produce and upload artifacts.

2) Subsequent jobs in the workflow download the artifacts, and use them.

I'm looking for the easiest way I can make changes to failed runs then restart them. The closet thing I can think of: manually download the artifacts that were created before the workflow failed, commit them to the repository, and have the initial jobs check if the artifacts the jobs were going to create are already committed in the checked-out repository, and short-circuit everything the jobs do to produce them; instead they just get uploaded directly in the new workflow run. But this feels messy.

**Comment by jsoref** [comment]
You could cheat and have a [`workflow_dispatch`](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow) that takes the artifact references as arguments, or a workflow run as an argument, and then retrieve things based on that...

**Reply by svarshavchik** [reply]
I'll have to look into that. If it's possible to take that argument and determine which ones of the previous workflow steps have failed, and retrieve the completed artifacts from the previous workflow, then this will work.

**Reply by jsoref** [reply]
You can definitely determine which steps failed.

Here's the code I use to look at stuff like this:
https://github.com/check-spelling/check-spelling/blob/c35bc3130c33b27843dd0b0f36be8f1d00b6d126/unknown-words.sh#L1436-L1459

Note that in my case, I'm looking for pending, but that's about the only difference.

**Comment by lobsterkatie** [comment]
My solution to this problem: The cache key for our artifacts is a variable in our workflow file. (We use commit SHA.) I just set that variable to a constant, commit the change, and then push. This kicks off one final run of artifact-creation, but after that, the artifacts are retrieved from the cache rather than rebuilt each time. Once everything is good I just rebase to drop that cache-key-change commit.

Here's the relevant yaml: https://github.com/getsentry/sentry-javascript/blob/5c462f1cec04ae2b925879095e6c5d011b37cee1/.github/workflows/build.yml#L121-L127
